src/commands/logs.rs

```diff
@@ -0,0 +1,168 @@
+
+    /// interval in secs to refresh logs from cloud
+    #[clap(parse(try_from_str = parse_interval), name="interval", long="interval", default_value = "2")]
+    pub interval_secs: u64,
+
+    /// fetch logs since
+    #[clap(parse(try_from_str = parse_duration), name="since", long="since", default_value = "7d")]
+    pub since: std::time::Duration,
+}
+
+impl LogsCommand {
+    pub async fn run(self) -> Result<()> {
+        let client = create_cloud_client(self.deployment_env_id.as_deref()).await?;
+        let app_name: String = self.app_name.clone();
+
+        self.logs(client, app_name.as_str())
+            .await
+            .map_err(|e| anyhow!("{:?}\n\nLearn more at {}", e, "DEVELOPER_CLOUD_FAQ"))
+    }
+
+    async fn logs(self, client: CloudClient, app_name: &str) -> Result<()> {
+        let app_id = match client
+            .get_app_id(app_name)
+            .await
+            .map_err(|_e| anyhow!("app with name {:?} not found", app_name))?
+        {
+            Some(x) => x,
+            None => return Err(anyhow!("app with name {:?} not found", app_name)),
+        };
+
+        let channel_id = client
+            .get_channel_id(app_id, SPIN_DEPLOY_CHANNEL_NAME)
+            .await?;
+
+        fetch_logs_and_print_loop(
+            &client,
+            channel_id,
+            self.follow,
+            self.interval_secs,
+            Some(self.max_lines),
+            self.since,
+        )
+        .await?;
+
+        Ok(())
+    }
+}
+
+async fn fetch_logs_and_print_loop(
+    client: &CloudClient,
+    channel_id: Uuid,
+    follow: bool,
+    interval: u64,
+    max_lines: Option<i32>,
+    since: Duration,
+) -> Result<()> {
+    let mut new_since = Utc::now().sub(since).to_rfc3339();
+    new_since =
+        fetch_logs_and_print_once(client, channel_id, max_lines, new_since.to_owned()).await?;
+
+    if !follow {
+        return Ok(());
+    }
+
+    loop {
+        tokio::time::sleep(tokio::time::Duration::from_secs(interval)).await;
+        new_since =
+            fetch_logs_and_print_once(client, channel_id, None, new_since.to_owned()).await?;
+    }
+}
+
+async fn fetch_logs_and_print_once(
+    client: &CloudClient,
+    channel_id: Uuid,
+    max_lines: Option<i32>,
+    since: String,
+) -> Result<String> {
+    let entries = client
+        .channel_logs_raw(channel_id.to_string(), max_lines, Some(since.to_string()))
+        .await?
+        .entries;
+
+    if entries.is_empty() {
+        return Ok(since.to_owned());
+    }
+
+    Ok(print_lastn_logs(&entries).to_owned())
+}
+
+fn print_lastn_logs(entries: &[Entry]) -> &str {
+    let mut new_since: &str = "";
+    for entry in entries.iter().rev() {
+        for line in entry.log_lines.as_ref().unwrap() {
+            println!("{}", line.line.as_ref().unwrap());
+            new_since = line.time.as_ref().unwrap().as_str()
+        }
+    }
+
+    new_since
+}
+
+fn parse_duration(arg: &str) -> Result<std::time::Duration, anyhow::Error> {
```

Comment:
tiny nit: I find this form more concise:

```suggestion
fn parse_duration(arg: &str) -> anyhow::Result<std::time::Duration> {
```